modules/carto/src/api/maps-api-client.js

```diff
@@ -84,83 +75,50 @@ function dealWithWindshaftError({response, layergroup, credentials}) {
           credentials.apiKey
         }') doesn't provide access to the requested data`
       );
+
     default:
-      throw new Error(`${JSON.stringify(layergroup.errors)}`);
+      const e = getMapsVersion() === 'v1' ? JSON.stringify(json.errors) : json.error;
+      throw new Error(e);
   }
 }
 
 /**
- * Create a GET or POST request, with all required parameters
+ * Build a URL all required parameters
  */
-function createMapsApiRequest({config, credentials}) {
+function buildURL({mapConfig, credentials}) {
+  const cfg = JSON.stringify(mapConfig);
   const encodedApiKey = encodeParameter('api_key', credentials.apiKey);
   const encodedClient = encodeParameter('client', `deck-gl-carto`);
   const parameters = [encodedApiKey, encodedClient];
-  const url = generateMapsApiUrl(parameters, credentials);
-
-  const getUrl = `${url}&${encodeParameter('config', config)}`;
-  if (getUrl.length < REQUEST_GET_MAX_URL_LENGTH) {
-    return getRequest(getUrl);
-  }
-
-  return postRequest(url, config);
-}
-
-/**
- * Generate a Maps API url for the request
- */
-function generateMapsApiUrl(parameters, credentials) {
-  const base = `${serverURL(credentials)}api/v1/map`;
-  return `${base}?${parameters.join('&')}`;
+  return `${mapsUrl(credentials)}?${parameters.join('&')}&${encodeParameter('config', cfg)}`;
 }
 
 /**
  * Prepare a url valid for the specified user
  */
-function serverURL(credentials) {
-  let url = credentials.serverUrlTemplate.replace(
-    DEFAULT_USER_COMPONENT_IN_URL,
-    credentials.username
-  );
-
-  if (!url.endsWith('/')) {
-    url += '/';
-  }
-
-  return url;
+function mapsUrl(credentials) {
+  return credentials.mapsUrl
+    .replace(DEFAULT_USER_COMPONENT_IN_URL, credentials.username)
+    .replace(DEFAULT_REGION_COMPONENT_IN_URL, credentials.region);
 }
 
 /**
- * Simple GET request
+ * Simple encode parameter
  */
-function getRequest(url) {
-  /* global Request */
-  /* eslint no-undef: "error" */
-  return new Request(url, {
-    method: 'GET',
-    headers: {
-      Accept: 'application/json'
-    }
-  });
+function encodeParameter(name, value) {
+  return `${name}=${encodeURIComponent(value)}`;
 }
 
 /**
- * Simple POST request
+ *  This function will be deprecated really soon when we'll make v2 the default
  */
-function postRequest(url, payload) {
-  return new Request(url, {
-    method: 'POST',
+async function getTileJSONBigQueryCloudFunction(mapConfig) {
+  const BQ_TILEJSON_ENDPOINT = 'https://us-central1-cartobq.cloudfunctions.net/tilejson';
```

Comment:
Does it make sense to make this url also configurable?

Reply:
Nope, because it won't be deprecated. The new endpoint for bqtiler layers is the Maps API, so it'll be configurable via config.mapsUrl

This code will be removed as soon as we'll do the switch to v2.